# 从经济因素（人均GDP，商品房房价，单元人员年平均工资）与教育因素（本科毕业人数）分析中国分省的结婚人数减少（恐婚现象）

# 数据：中国分省近十年来的人均GDP，商品房房价，城镇单位人员年平均工资，本科毕业人数，以及结婚登记人数

# 数据来源：国家统计局

In [147]:
import pandas as pd
df = pd.read_csv('C:/Users/Lenovo/Desktop/sum.csv')
df

province   GDP18   GDP17   GDP16   GDP15   GDP14   GDP13  GDP12  GDP11  \
0        北京市  140211  128994  118198  106497   99995   94648  87475  81658   
1        天津市  120711  118944  115053  107960  105231  100105  93173  85213   
2        河北省   47772   45387   43062   40255   39984   38909  36584  33969   
3        山西省   45328   42060   35532   34919   35070   34984  33628  31357   
4     内蒙古自治区   68302   63764   72064   71101   71046   67836  63886  57974   
5        辽宁省   58008   53527   50791   65354   65201   61996  56649  50760   
6        吉林省   55611   54838   53868   51086   50160   47428  43415  38460   
7       黑龙江省   43274   41916   40432   39462   39226   37697  35711  32819   
8        上海市  134982  126634  116562  103796   97370   90993  85373  82560   
9        江苏省  115168  107150   96887   87995   81874   75354  68347  62290   
10       浙江省   98643   92057   84916   77644   73002   68805  63374  59249   
11       安徽省   47712   43401   39561   35997   34425   32001  28792  25659   
12       福建省   91197   82677   74707   67966   63472   58145  52763  47377   
13       江西省   47434   43424   40400   36724   34674   31930  28800  26150   
14       山东省   76267   72807   68733   64168   60879   56885  51768  47335   
15       河南省   50152   46674   42575   39123   37072   34211  31499  28661   
16       湖北省   66616   60199   55665   50654   47145   42826  38572  34197   
17       湖南省   52949   49558   46382   42754   40271   36943  33480  29880   
18       广东省   86412   80932   74016   67503   63469   58833  54095  50807   
19   广西壮族自治区   41489   38102   38027   35190   33090   30741  27952  25326   
20       海南省   51955   48430   44347   40818   38924   35663  32377  28898   
21       重庆市   65933   63442   58502   52321   47850   43223  38914  34500   
22       四川省   48883   44651   40003   36775   35128   32617  29608  26133   
23       贵州省   41244   37956   33246   29847   26437   23151  19710  16413   
24       云南省   37136   34221   31093   28806   27264   25322  22195  19265   
25     西藏自治区   43398   39267   35184   31999   29252   26326  22936  20077   
26       陕西省   63477   57266   51015   47626   46929   43117  38564  33464   
27       甘肃省   31336   28497   27643   26165   26433   24539  21978  19595   
28       青海省   47689   44047   43531   41252   39671   36875  33181  29522   
29   宁夏回族自治区   54094   50765   47194   43805   41834   39613  36394  33043   
30  新疆维吾尔自治区   49475   44941   40564   40036   40648   37553  33796  30087   

    GDP10    ...     salary18  salary17  salary16  salary15  salary14  \
0   73856    ...       145766    131700    119928    111390    102268   
1   72994    ...       100731     94534     86305     80090     72773   
2   28668    ...        68717     63036     55334     50921     45114   
3   26283    ...        65917     60061     53705     51803     48969   
4   47347    ...        73835     66679     61067     57135     53748   
5   42355    ...        67324     61153     56015     52332     48190   
6   31599    ...        68533     61451     56098     51558     46516   
7   27076    ...        60780     56067     52435     48881     44036   
8   76074    ...       140400    129795    119935    109174    100251   
9   52840    ...        84688     78267     71574     66196     60867   
10  51711    ...        88883     80750     73326     66668     61572   
11  20888    ...        74378     65150     59102     55139     50894   
12  40025    ...        74316     67420     61973     57628     53426   
13  21253    ...        68573     61429     56136     50932     46218   
14  41106    ...        73593     68081     62539     57270     51825   
15  24446    ...        63174     55495     49505     45403     42179   
16  27906    ...        73777     65912     59831     54367     49838   
17  24719    ...        70221     63690     58241     52357     47117   
18  44736    ...        88636     79183     72326     65788     59481   
19  20219    ...        70606     63821     578

In [68]:
df.columns

Index(['province', 'GDP18', 'GDP17', 'GDP16', 'GDP15', 'GDP14', 'GDP13',
       'GDP12', 'GDP11', 'GDP10', 'GDP09', 'marry18', 'marry17', 'marry16',
       'marry15', 'marry14', 'marry13', 'marry12', 'marry11', 'marry10',
       'marry09', 'price18', 'price17', 'price16', 'price15', 'price14',
       'price13', 'price12', 'price11', 'price10', 'price09', 'edu18', 'edu17',
       'edu16', 'edu15', 'edu14', 'edu13', 'edu12', 'edu11', 'edu10', 'edu09',
       'salary18', 'salary17', 'salary16', 'salary15', 'salary14', 'salary13',
       'salary12', 'salary11', 'salary10', 'salary09'],
      dtype='object')

In [2]:
print(list(df.province))

['北京市', '天津市', '河北省', '山西省', '内蒙古自治区', '辽宁省', '吉林省', '黑龙江省', '上海市', '江苏省', '浙江省', '安徽省', '福建省', '江西省', '山东省', '河南省', '湖北省', '湖南省', '广东省', '广西壮族自治区', '海南省', '重庆市', '四川省', '贵州省', '云南省', '西藏自治区', '陕西省', '甘肃省', '青海省', '宁夏回族自治区', '新疆维吾尔自治区']


In [3]:
print(list(df.marry18))

[13.78, 9.75, 45.87, 27.85, 17.69, 28.08, 19.32, 27.82, 10.51, 63.77, 33.73, 61.91, 27.36, 33.06, 59.9, 80.9, 43.78, 42.16, 71.38, 36.19, 7.27, 25.88, 66.9, 40.03, 37.76, 3.01, 30.04, 21.07, 5.73, 6.07, 15.37]


In [4]:
print(list(df.marry09))

[18.18, 10.4, 72.0, 34.36, 18.84, 38.17, 23.91, 30.39, 14.99, 73.09, 42.76, 61.0, 36.06, 40.81, 91.84, 85.88, 56.54, 65.29, 86.45, 51.56, 13.0, 30.11, 71.0, 30.79, 34.29, 1.36, 31.94, 12.24, 2.97, 5.66, 26.55]


In [69]:
print(list(df.salary09))

[57779, 43937, 27774, 28066, 30486, 30523, 25943, 24805, 58336, 35217, 36553, 28723, 28366, 24165, 29398, 26906, 26547, 26534, 36469, 27322, 24790, 30499, 28149, 27437, 26163, 45347, 29566, 26743, 32481, 32916, 27617]


In [70]:
print(list(df.salary18))

[145766, 100731, 68717, 65917, 73835, 67324, 68533, 60780, 140400, 84688, 88883, 74378, 74316, 68573, 73593, 63174, 73777, 70221, 88636, 70606, 75885, 78928, 77686, 78316, 75701, 116015, 71983, 70695, 85379, 78384, 75457]


In [124]:
from pyecharts.charts import Map,Geo
from pyecharts import options as opts
#将数据处理成列表
locate = ['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆']

marry09y = [18.18, 10.4, 72.0, 34.36, 18.84, 38.17, 23.91, 30.39, 14.99, 73.09, 42.76, 61.0, 36.06, 40.81, 91.84, 85.88, 56.54, 65.29, 86.45, 51.56, 13.0, 30.11, 71.0, 30.79, 34.29, 1.36, 31.94, 12.24, 2.97, 5.66, 26.55]

list1 = [[locate[i],marry09y[i]] for i in range(len(locate))]

map_1 = Map()

map_1.set_global_opts(
    title_opts=opts.TitleOpts(title="2009年中国各省结婚人数数值"), 
    visualmap_opts=opts.VisualMapOpts(max_=55)  #最大数据范围
    )
map_1.add("2009年结婚人数（单位：万人）", list1, maptype="china")
map_1.render_notebook()

In [125]:
from pyecharts.charts import Map,Geo
from pyecharts import options as opts
#将数据处理成列表
locate = ['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆']

marry18y = [13.78, 9.75, 45.87, 27.85, 17.69, 28.08, 19.32, 27.82, 10.51, 63.77, 33.73, 61.91, 27.36, 33.06, 59.9, 80.9, 43.78, 42.16, 71.38, 36.19, 7.27, 25.88, 66.9, 40.03, 37.76, 3.01, 30.04, 21.07, 5.73, 6.07, 15.37]

list2 = [[locate[i],marry18y[i]] for i in range(len(locate))]

map_2 = Map()

map_2.set_global_opts(
    title_opts=opts.TitleOpts(title="2018年中国各省结婚人数数值"), 
    visualmap_opts=opts.VisualMapOpts(max_=55)  #最大数据范围
    )
map_2.add("2018年结婚人数（单位：万人）", list2, maptype="china")
map_2.render_notebook()

- 从两个中国地图中2009年与2018年的分省结婚人数对比中我们可以看到，在中东部的地区的结婚人数变化最明显，在我们的认知中，中东部地区的发展比起西部地区发展更为发达，我们的初步推算是，经济发展是影响结婚人数的一个重要的因数。

- 后来我们认为仅从经济发展来分析结婚人数的减少是不够全面的，那么我们增加了以下几个变量：分省GDP,城镇单位人员的平均工资，本科毕业人数，以及商品房的平均价格。

In [148]:
print(list(df.GDP18))

[140211, 120711, 47772, 45328, 68302, 58008, 55611, 43274, 134982, 115168, 98643, 47712, 91197, 47434, 76267, 50152, 66616, 52949, 86412, 41489, 51955, 65933, 48883, 41244, 37136, 43398, 63477, 31336, 47689, 54094, 49475]


In [149]:
print(list(df.GDP09))

[66940, 62574, 24581, 21522, 39735, 35149, 26595, 22447, 69165, 44253, 43842, 16408, 33437, 17335, 35894, 20597, 22677, 20428, 39436, 16045, 19254, 22920, 17339, 10971, 13539, 15008, 21947, 13269, 19454, 21777, 19942]


In [13]:
print(list(df.price18))

[37420.19, 15924.26, 7566.72, 6649.11, 5339.79, 7358.35, 6800.56, 6678.06, 28981.11, 10542.48, 15242.01, 6937.06, 10612.64, 6539.84, 7386.25, 5530.6, 8135.76, 5473.14, 12915.25, 5958.53, 14104.76, 8189.98, 6691.23, 5129.19, 7384.44, 6914.73, 7921.85, 5387.05, 5936.6, 4736.94, 5427.31]


In [14]:
print(list(df.price09))

[13224, 6605, 3210, 2552, 2649, 3872, 2788, 3067, 12364, 4805, 7890, 3235, 5366, 2517, 3390, 2501, 3413, 2532, 6360, 3133, 6291, 3266, 3434, 2642, 2723, 2392, 3113, 2396, 2442, 2824, 2466]


In [29]:
print(list(df.edu09))

[11.21, 5.34, 9.95, 5.78, 3.08, 12.58, 8.82, 9.44, 7.19, 18.68, 9.9, 8.27, 6.62, 8.15, 18.0, 12.77, 15.01, 10.62, 13.45, 4.24, 1.06, 6.12, 12.5, 3.07, 4.28, 0.47, 10.9, 3.96, 0.55, 0.77, 2.77]


In [30]:
print(list(df.edu18))

[12.03, 7.85, 17.14, 11.75, 5.8, 17.75, 11.31, 12.49, 8.58, 26.01, 14.93, 15.41, 12.1, 12.69, 23.96, 26.2, 20.4, 16.42, 25.4, 9.75, 2.49, 10.53, 18.64, 6.99, 9.57, 0.56, 16.96, 7.06, 0.83, 1.79, 3.45]


In [129]:
print(list(df.salary18))

[145766, 100731, 68717, 65917, 73835, 67324, 68533, 60780, 140400, 84688, 88883, 74378, 74316, 68573, 73593, 63174, 73777, 70221, 88636, 70606, 75885, 78928, 77686, 78316, 75701, 116015, 71983, 70695, 85379, 78384, 75457]


In [130]:
print(list(df.salary09))

[57779, 43937, 27774, 28066, 30486, 30523, 25943, 24805, 58336, 35217, 36553, 28723, 28366, 24165, 29398, 26906, 26547, 26534, 36469, 27322, 24790, 30499, 28149, 27437, 26163, 45347, 29566, 26743, 32481, 32916, 27617]


In [150]:
from pyecharts import options as opts
from pyecharts.charts import Bar, Grid, Line,Scatter


def grid_vertical() -> Grid:
    bar= (
        Bar()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2018年分省人均GDP",[140211, 120711, 47772, 45328, 68302, 58008, 55611, 43274, 134982, 115168, 98643, 47712, 91197, 47434, 76267, 50152, 66616, 52949, 86412, 41489, 51955, 65933, 48883, 41244, 37136, 43398, 63477, 31336, 47689, 54094, 49475])
        .add_yaxis("2009年分省人均GDP", [66940, 62574, 24581, 21522, 39735, 35149, 26595, 22447, 69165, 44253, 43842, 16408, 33437, 17335, 35894, 20597, 22677, 20428, 39436, 16045, 19254, 22920, 17339, 10971, 13539, 15008, 21947, 13269, 19454, 21777, 19942]
)
        .set_global_opts(title_opts=opts.TitleOpts(title="2018年与2009年分省GDP对比（单位：万元）"))
    )
    line = (
        Line()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2018年分省房价", [37420.19, 15924.26, 7566.72, 6649.11, 5339.79, 7358.35, 6800.56, 6678.06, 28981.11, 10542.48, 15242.01, 6937.06, 10612.64, 6539.84, 7386.25, 5530.6, 8135.76, 5473.14, 12915.25, 5958.53, 14104.76, 8189.98, 6691.23, 5129.19, 7384.44, 6914.73, 7921.85, 5387.05, 5936.6, 4736.94, 5427.31])
        .add_yaxis("2009年分省房价", [13224, 6605, 3210, 2552, 2649, 3872, 2788, 3067, 12364, 4805, 7890, 3235, 5366, 2517, 3390, 2501, 3413, 2532, 6360, 3133, 6291, 3266, 3434, 2642, 2723, 2392, 3113, 2396, 2442, 2824, 2466])
        .set_global_opts(
            title_opts=opts.TitleOpts(title="2018年与2009年分省房价对比（单位：元）", pos_top="48%"),
            legend_opts=opts.LegendOpts(pos_top="48%"),
        )
    )

    grid = (
        Grid()
        .add(bar, grid_opts=opts.GridOpts(pos_bottom="60%",pos_right="0",height="30%"))
        .add(line, grid_opts=opts.GridOpts(pos_top="60%",pos_right="0",height="30%"))
    )
    return grid

a = grid_vertical()
a.render_notebook()

- 在上图中的2009年与2018年的分省人均GDP与房价的对比，从图中我们可以看到人均GDP增长速度最快的是北京，上海，江苏，山东，广东等地区，而这些地区也是在近十年来结婚人数变化趋势较为明显的地区。在房价对比的折线图中我们可以看到，大部分的地区房价变化是没有太大的波动的，都是趋于平稳地上涨，但有两个地方的放假上涨的幅度最为明显，那就是北京与上海，他房价变化大，主要是他区域的面积小，但经济发达，寸土寸金。但我们从地图中可以看到，其实北京与上海的结婚人数变化其实没有太大的变化趋势，我们结合我们的背景，可以知道，北上广是一个经济发达，许多内地的人们都去北上广闯荡希望能够赚取更多的钱，但这些人很少会定居在北上广，通常都会回到自己的家乡定居，这就能够说明北京与上海的结婚人数变化不明显，但房价变化很大的原因。

In [151]:
from pyecharts import options as opts
from pyecharts.charts import Bar, Grid, Line,Scatter


def grid_vertical() -> Grid:
    bar= (
        Bar()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2009年分省单位人员平均工资（单位：元）",[57779, 43937, 27774, 28066, 30486, 30523, 25943, 24805, 58336, 35217, 36553, 28723, 28366, 24165, 29398, 26906, 26547, 26534, 36469, 27322, 24790, 30499, 28149, 27437, 26163, 45347, 29566, 26743, 32481, 32916, 27617])
        .add_yaxis("2009年分省商品房房价（单位：元/㎡）", [13224, 6605, 3210, 2552, 2649, 3872, 2788, 3067, 12364, 4805, 7890, 3235, 5366, 2517, 3390, 2501, 3413, 2532, 6360, 3133, 6291, 3266, 3434, 2642, 2723, 2392, 3113, 2396, 2442, 2824, 2466]
)
        .set_global_opts(title_opts=opts.TitleOpts(title="2009年分省平均工资与房价对比"))
    )
    line = (
        Line()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2018年分省单位人员平均工资（单位：元）", [145766, 100731, 68717, 65917, 73835, 67324, 68533, 60780, 140400, 84688, 88883, 74378, 74316, 68573, 73593, 63174, 73777, 70221, 88636, 70606, 75885, 78928, 77686, 78316, 75701, 116015, 71983, 70695, 85379, 78384, 75457])
        .add_yaxis("2018年分省商品房房价（单位：元/㎡）", [37420.19, 15924.26, 7566.72, 6649.11, 5339.79, 7358.35, 6800.56, 6678.06, 28981.11, 10542.48, 15242.01, 6937.06, 10612.64, 6539.84, 7386.25, 5530.6, 8135.76, 5473.14, 12915.25, 5958.53, 14104.76, 8189.98, 6691.23, 5129.19, 7384.44, 6914.73, 7921.85, 5387.05, 5936.6, 4736.94, 5427.31])
        .set_global_opts(
            title_opts=opts.TitleOpts(title="2018年分省平均工资与房价对比", pos_top="48%"),
            legend_opts=opts.LegendOpts(pos_top="48%"),
        )
    )

    grid = (
        Grid()
        .add(bar, grid_opts=opts.GridOpts(pos_bottom="60%",pos_right="0",height="30%"))
        .add(line, grid_opts=opts.GridOpts(pos_top="60%",pos_right="0",height="30%"))
    )
    return grid

a = grid_vertical()
a.render_notebook()

- 在图中我们可以看出2009年的时候，除了一些经济非常发达的地区，如北京，上海等地区房价商品房的房价上万以外，其他地区的房价基本上都是在2000元-3000元/平方米范围之内，而在2009年的时候排除了一些经济非常发达的地区，人们的年平均工资基本上都是在25000-35000元的区域之内，这时我们推算一个家庭两个人的一年的平均工资为70000元，而商品房的房价价格为2000元左右的价格，那么他们的供房的困难还是较为少的，而且内地地区在十年前经济较为不发达，多为自建的住宅区，那么他们的住房房价更为便宜，结婚的压力也没有这么大。在2018年的时候，平均工资上涨了，基本上是70000-80000元的区域，但商品房的房价也随之上涨，基本上都是在6000元以上一平方米，虽然工资上涨了，但是经济发展，物价，人们对生活的要求提高（如精神上的需求，旅游等），那么供房的难度也大了。

In [145]:
from pyecharts import options as opts
from pyecharts.charts import Bar, Grid, Line,Scatter


def grid_vertical() -> Grid:
    bar= (
        Bar()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2009年分省结婚人数（单位：万对）",[18.18, 10.4, 72.0, 34.36, 18.84, 38.17, 23.91, 30.39, 14.99, 73.09, 42.76, 61.0, 36.06, 40.81, 91.84, 85.88, 56.54, 65.29, 86.45, 51.56, 13.0, 30.11, 71.0, 30.79, 34.29, 1.36, 31.94, 12.24, 2.97, 5.66, 26.55])
        .add_yaxis("2018年分省结婚人数（单位：万对）", [13.78, 9.75, 45.87, 27.85, 17.69, 28.08, 19.32, 27.82, 10.51, 63.77, 33.73, 61.91, 27.36, 33.06, 59.9, 80.9, 43.78, 42.16, 71.38, 36.19, 7.27, 25.88, 66.9, 40.03, 37.76, 3.01, 30.04, 21.07, 5.73, 6.07, 15.37]
)
        .set_global_opts(title_opts=opts.TitleOpts(title="2009年与2018年分省结婚人数对比"))
    )
    line = (
        Line()
        .add_xaxis(['北京', '天津', '河北', '山西', '内蒙古', '辽宁', '吉林', '黑龙江', '上海', '江苏', '浙江', '安徽', '福建', '江西', '山东', '河南', '湖北', '湖南', '广东', '广西', '海南', '重庆', '四川', '贵州', '云南', '西藏', '陕西', '甘肃', '青海', '宁夏', '新疆'])
        .add_yaxis("2009年分省本科毕业人数（单位：万人）", [11.21, 5.34, 9.95, 5.78, 3.08, 12.58, 8.82, 9.44, 7.19, 18.68, 9.9, 8.27, 6.62, 8.15, 18.0, 12.77, 15.01, 10.62, 13.45, 4.24, 1.06, 6.12, 12.5, 3.07, 4.28, 0.47, 10.9, 3.96, 0.55, 0.77, 2.77])
        .add_yaxis("2018年分省本科毕业人数（单位：万人）", [12.03, 7.85, 17.14, 11.75, 5.8, 17.75, 11.31, 12.49, 8.58, 26.01, 14.93, 15.41, 12.1, 12.69, 23.96, 26.2, 20.4, 16.42, 25.4, 9.75, 2.49, 10.53, 18.64, 6.99, 9.57, 0.56, 16.96, 7.06, 0.83, 1.79, 3.45])
        .set_global_opts(
            title_opts=opts.TitleOpts(title="2009年与2018年分省本科毕业人数对比", pos_top="48%"),
            legend_opts=opts.LegendOpts(pos_top="48%"),
        )
    )

    grid = (
        Grid()
        .add(bar, grid_opts=opts.GridOpts(pos_bottom="60%",pos_right="0",height="30%"))
        .add(line, grid_opts=opts.GridOpts(pos_top="60%",pos_right="0",height="30%"))
    )
    return grid

a = grid_vertical()
a.render_notebook()

- 在上图中我们可以看到，总体上结婚人数都是呈下降的趋势的，有的变化幅度大，如山东，湖南等地区，下降的幅度较大；有的变化幅度较小，甚至是上升涨的趋势的是西藏，青海等地区。在本科毕业人数中我们可以看出，大部分的地区都是相对于2009年本科的毕业人数都是有所上涨的，有些地区的变化幅度甚至在十万人以上，如河南省等，这说明了随着经济的发展，国家对教育的重视程度提高，使高校的毕业人数增加。

- 但为什么河南的教育程度提高，而结婚人数无论是2009年还是2018年都居于高位？根据资料现在那是因为河南受传统文化“先成家后立业”的影响较为深远，同时河南嫁娶的彩礼较低等原因导致。





# 我们抽取结婚人数变化幅度最大的山东以及变化幅度最小的宁夏分析

###  山东省的恐婚因素分析

In [193]:
#from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Bar, Tab, Pie, Line
from pyecharts.components import Table


def bar_datazoom_slider() -> Bar:
    c = (
        Bar()
        .add_xaxis(['2009年', '2010年', '2011年', '2012年', '2013年', '2014年', '2015年', '2016年', '2017年', '2018年'])
        .add_yaxis("山东近十年来商品房房价（元/平方米）",[3390, 3809, 4298.78, 4556.63, 4797, 5029, 5290, 5728, 6153, 7386.25]) 
        .add_yaxis("山东近十年来城镇单位人员平均工资（元）",[29398, 33321, 37618, 41904, 46998, 51825, 57270, 62539, 68081, 73593]) 
        .set_global_opts(
            title_opts=opts.TitleOpts(title="山东省商品房房价与平均工资对比"),

            datazoom_opts=[opts.DataZoomOpts()],
        )
    )
    return c


def line_markpoint() -> Line:
    c = (
        Line()
        .add_xaxis(['2009年', '2010年', '2011年', '2012年', '2013年', '2014年', '2015年', '2016年', '2017年', '2018年'])
        .add_yaxis("山东近十年来结婚人数",[91.84, 92.5, 96.91, 93.33, 89.14, 83.18, 70.1, 67.07, 62.58, 59.9],
            markpoint_opts=opts.MarkPointOpts(data=[opts.MarkPointItem(type_="min")]),
        )
        .add_yaxis("山东近十年来本科毕业毕业人数",[18, 18.27, 17.93, 20.56, 21.17, 21.55, 22.75, 23.66, 24.55, 23.96],
            markpoint_opts=opts.MarkPointOpts(data=[opts.MarkPointItem(type_="max")]),
        )
        .set_global_opts(title_opts=opts.TitleOpts(title="近十年来结婚人数与本科毕业人数（单位：万人）"))
    )
    return c



def table_base() -> Table:
    table = Table()

    headers = ["年份", "2009", "2010", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018"]
    rows = [
        ["人均GDP（元）", 35894, 41106, 47335, 51768, 56885, 60879, 64168, 68733, 72807, 76267],
        ["商品房房价（元）", 3390, 3809, 4298.78, 4556.63, 4797, 5029, 5290, 5728, 6153, 7386.25],
        ["城镇单位人员平均工资（元）", 29398, 33321, 37618, 41904, 46998, 51825, 57270, 62539, 68081, 73593],
        ["结婚人数（万人）", 91.84, 92.5, 96.91, 93.33, 89.14, 83.18, 70.1, 67.07, 62.58, 59.9],
        ["本科毕业人数（万人）",18, 18.27, 17.93, 20.56, 21.17, 21.55, 22.75, 23.66, 24.55, 23.96],
#        ["Sydney", 2058, 4336374, 1214.8],
#        ["Melbourne", 1566, 3806092, 646.9],
#        ["Perth", 5386, 1554769, 869.4],
    ]
    table.add(headers, rows).set_global_opts(
        title_opts=opts.ComponentTitleOpts(title="山东省近十年来的相关数据")
    )
    return table

tab = Tab()
tab.add(bar_datazoom_slider(), "bar-example")
tab.add(line_markpoint(), "line-example")
#tab.add(bar_datazoom_slider(), "bar-example")
tab.add(table_base(), "table-example")
tab.render_notebook()

- 第一，在条形图中我们可以看出，近十年来山东的商品房房价在不断地上涨，从3390元/平方米，上涨至7386元/平方米，这是差距非常大的一个数据，一平方米相差4000元，那么如果一对即将需要买房的夫妻来说，如果他们要买50平方米，相对来说是差距20万元。从山东省单位人员的平均工资来看，虽然从2009年到2018年年平均工资上涨了4万元，但我们从每一年来看，他的上升幅度是很少的，平均每年上升4000元左右，但对于买房来说，压力也是非常大的。

- 第二，在山东省的结婚人数折线图中我们可以看出，在2009年至2011年期间，结婚人数是呈上升的趋势的，而2011年达到顶峰，根据搜集资料发现，在2011年推行试行了“二胎政策”，这可能会吸引了的结婚导致结婚人数上涨。在2011年后，结婚人数有着快速下滑的趋势，从96.91万对下降到2018年的59.9万对。根据搜集资料发现，山东因为转型升级，劳动密集型产业劳动人员减少，导致相应的结婚人数也减少。随着教育的不断发展，山东的本科毕业人数也在缓慢地增长。受教育程度的提高，婚姻观的变化也是影响结婚人数的一个重要因数。

## 宁夏回族自治区的恐婚因素分析

In [197]:
#from pyecharts.faker import Faker
from pyecharts import options as opts
from pyecharts.charts import Bar, Tab, Pie, Line
from pyecharts.components import Table


def bar_datazoom_slider() -> Bar:
    c = (
        Bar()
        .add_xaxis(['2009年', '2010年', '2011年', '2012年', '2013年', '2014年', '2015年', '2016年', '2017年', '2018年'])
        .add_yaxis("宁夏近十年来人均GDP（元）",[21777, 26860, 33043, 36394, 39613, 41834, 43805, 47194, 50765, 54094]) 
#        .add_yaxis("山东近十年来城镇单位人员平均工资（元）",[29398, 33321, 37618, 41904, 46998, 51825, 57270, 62539, 68081, 73593]) 
        .set_global_opts(
            title_opts=opts.TitleOpts(title="近十年来宁夏人均GDP"),

            datazoom_opts=[opts.DataZoomOpts()],
        )
    )
    return c


def line_markpoint() -> Line:
    c = (
        Line()
        .add_xaxis(['2009年', '2010年', '2011年', '2012年', '2013年', '2014年', '2015年', '2016年', '2017年', '2018年'])
        .add_yaxis("宁夏近十年来结婚人数",[5.66, 6, 6.15, 6.07, 6.38, 5.6, 6.09, 6.2, 6.03, 6.07],
            markpoint_opts=opts.MarkPointOpts(data=[opts.MarkPointItem(type_="min")]),
        )
        .add_yaxis("宁夏近十年来本科毕业毕业人数",[0.77, 0.92, 1, 1.18, 1.26, 1.33, 1.55, 1.68, 1.72, 1.79],
            markpoint_opts=opts.MarkPointOpts(data=[opts.MarkPointItem(type_="max")]),
        )
        .set_global_opts(title_opts=opts.TitleOpts(title="近十年来结婚人数与本科毕业人数（单位：万人）"))
    )
    return c



#def table_base() -> Table:
#    table = Table()

#    headers = ["年份", "2009", "2010", "2011", "2012", "2013", "2014", "2015", "2016", "2017", "2018"]
#    rows = [
#        ["人均GDP（万元）", 33896.65, 39169.92, 45361.85, 50013.24, 55230.32, 59426.59, 63002.33, 68024.49, 72634.15, 76469.67],
#        ["商品房房价（元）", 3390, 3809, 4298.78, 4556.63, 4797, 5029, 5290, 5728, 6153, 7386.25],
#        ["城镇单位人员平均工资（元）", 29398, 33321, 37618, 41904, 46998, 51825, 57270, 62539, 68081, 73593],
#        ["结婚人数（万人）", 91.84, 92.5, 96.91, 93.33, 89.14, 83.18, 70.1, 67.07, 62.58, 59.9],
#        ["本科毕业人数（万人）",18, 18.27, 17.93, 20.56, 21.17, 21.55, 22.75, 23.66, 24.55, 23.96],
#        ["Sydney", 2058, 4336374, 1214.8],
#        ["Melbourne", 1566, 3806092, 646.9],
#        ["Perth", 5386, 1554769, 869.4],
#    ]
#    table.add(headers, rows).set_global_opts(
#        title_opts=opts.ComponentTitleOpts(title="山东省近十年来的相关数据")
#    )
#    return table

tab = Tab()
tab.add(bar_datazoom_slider(), "bar-example")
tab.add(line_markpoint(), "line-example")
#tab.add(bar_datazoom_slider(), "bar-example")
#tab.add(table_base(), "table-example")
tab.render_notebook()

- 从图表中我们可以看到，宁夏回族自治区在近十年来的人均GPD的增长速度相对于其他地区来说增长是相对来说较为缓慢的，这与它的地理位置，人口密度，政策等许多因素相关。第二个折线图中我们可以看到，宁夏的结婚人数变化不明显，近十年来结婚人数最高点与最低点相差一万人不够，因此可以推算出宁夏的结婚率是较为稳定的。在本科毕业人数的折线图中我们可以看到，本科的毕业人数是逐年上升的，我们可以得到宁夏地区的受教育程度不断地提高。人们的思想更加地开放，婚姻观也随之改变。但是因为宁夏地理位置以及经济发展的原因，人员流动较少，因此结婚人数的变化较不明显。

In [200]:
#from pyecharts.faker import Faker    变化最少宁夏
from pyecharts import options as opts
from pyecharts.charts import Bar, Tab, Pie, Line
from pyecharts.components import Table

def grid_mutil_yaxis() -> Grid:
    x_data = ["{}年".format(i) for i in range(2009, 2019)]
    bar = (
        Bar()
        .add_xaxis(x_data)
        .add_yaxis(
            "平均工资",
            [32916, 37166, 42703, 47436, 50476, 54858, 60380, 65570, 70298, 78384],
#            [2824, 3107, 3389.46, 3620.77, 3917, 3747, 4010, 3925, 4243, 4736.94],
            yaxis_index=0,
            color="#d14a61",
        )
        .add_yaxis(
            "房价",
            [2824, 3107, 3389.46, 3620.77, 3917, 3747, 4010, 3925, 4243, 4736.94],
#            [32916, 37166, 42703, 47436, 50476, 54858, 60380, 65570, 70298, 78384],
            yaxis_index=1,
            color="#5793f3",
        )
        .extend_axis(
            yaxis=opts.AxisOpts(
                name="房价（单位：元/㎡）",
                type_="value",
                min_=0,
                max_=100000,
                position="right",
                axisline_opts=opts.AxisLineOpts(
                    linestyle_opts=opts.LineStyleOpts(color="#d14a61")
                ),
                axislabel_opts=opts.LabelOpts(formatter="{value} 元/㎡"),
            )
        )
        .extend_axis(
            yaxis=opts.AxisOpts(
                type_="value",
                name="平均工资（单位：元）",
                min_=0,
#                max_=10000,
                position="left",
                axisline_opts=opts.AxisLineOpts(
                    linestyle_opts=opts.LineStyleOpts(color="#675bba")
                ),
                axislabel_opts=opts.LabelOpts(formatter="{value}元"),
                splitline_opts=opts.SplitLineOpts(
                    is_show=True, linestyle_opts=opts.LineStyleOpts(opacity=1)
                ),
            )
        )

    )
 
    


#    bar.overlap(line)
    return Grid().add(bar, opts.GridOpts(pos_left="10%", pos_right="10%"), is_control_axis_index=True)
a = grid_mutil_yaxis()
a.render_notebook()


- 在图中我们可以看到，近十年来宁夏的商品房房价其实变化的幅度较为小的，从2009年的2824元/平方米到2018年4736元/平方米，十年来每平方米增长了近2000元，增长幅度较少。而人们的平均工资也是呈上升的趋势，单位员工的平均工资增长速度较快，从2009年的32916元增长至2018年的78384元，这样子的年平均工资人们购房的难度较低，因此受到购房问题困扰的人群较少。

# 总结

近年来我国各省的结婚人数普遍呈下降的趋势，我们主要是通过经济方面（各省人均GDP，各省商品房房价，单元员工平均工资）以及受教育程度（本科毕业人数）来分析近十年来结婚人数减少的原因，剖析原因及其危害，并为其提出建议。

从以上的分析中我们可以看到，中国经济越发达地区的结婚率越低，这主要是因为经济发展，导致物价，房价等人们生活的成本提高，人口生活压力增大。同时结婚的成本增大，如一些“没有稳定的工作和收入不结婚”“没房没车不结婚”等都成了结婚的附加条件，与此同时，经济发展，经济发达的地区的房价不断地上升，导致购房成为了年轻人的一大难题。许多年轻人也因此成为“房奴”等。这些经济因素都会导致年轻人对结婚这个问题望而止步。同时，随着经济的发展，人们受教育的程度不断提高，人们的婚姻观念的改变，不仅是以“成家”为主，更多的倾向于“立业”，不断地充实自己，追求事业上的成功，不愿意早早地结婚束缚住自己的发展。
结婚人数不断地减少的社会经济发展进步的一个必然结果，但是结婚屡创新低，对社会发展也存在不利影响。一个国家的繁荣与发展，首先有赖于人口的稳定和发展，结婚率走低，带来的直接影响就是出生率的降低，以及人口代际的延长。比如，25岁结婚，26岁生育，百年之内基本上将产生四代人。但是35岁结婚，36岁生育，百年之内将只有三代人，时间越长，人口出生率越低，这就是现实。 而人口老龄化的问题也随之涌现，老年人口的抚养问题等对年轻人也是一种压力。

针对以上问题的建议是虽然经济问题是一个很大的压力，但是也不要因为经济的原因而耽误了自己的婚姻幸福，应树立正确的社会导向，把成家立业、生儿育女、快乐生活当作人生乐趣，形成婚姻家庭的良性循环。

